index.js

```diff
@@ -0,0 +1,214 @@
+    ['address[]', 'int256[]'],
+    [Object.keys(participants), Object.values(participants)]
+  )
+  const reqSigner = ethers.verifyMessage(
+    digest,
+    ethers.Signature.from(signature)
+  )
+  httpAssert(signerAddresses.includes(reqSigner), 403, 'Invalid signature')
+}
+
+async function handleIncreaseScores (req, res, redis, signerAddresses) {
+  const body = JSON.parse(await getRawBody(req, { limit: '1mb' }))
+
+  httpAssert(
+    typeof body === 'object' && body !== null,
+    400,
+    'Request body should be an object'
+  )
+  httpAssert(
+    typeof body.scores === 'object' && body.scores !== null,
+    400,
+    '.scores should be an object'
+  )
+  httpAssert(
+    Object.keys(body.scores).every(ethers.isAddress),
+    400,
+    'All .scores keys should be 0x addresses'
+  )
+  httpAssert(
+    Object.values(body.scores).every(score => {
+      try {
+        return BigInt(score) > 0n
+      } catch {
+        return false
+      }
+    }),
+    400,
+    'All .scores values should be positive numbers encoded as string'
+  )
+
+  validateSignature(body.signature, body.scores, signerAddresses)
+
+  const timestamp = new Date()
+  const tx = redis.multi()
+  for (const [address, score] of Object.entries(body.scores)) {
+    const scheduledRewards = (BigInt(score) * roundReward) / maxScore
+    tx.hincrby('rewards', address, scheduledRewards)
+    tx.rpush(
+      'log',
+      JSON.stringify({
+        timestamp,
+        address,
+        score,
+        scheduledRewards: String(scheduledRewards)
+      })
+    )
+  }
+  const updated = await tx.exec()
+
+  json(
+    res,
+    Object.fromEntries(
+      Object
+        .keys(body.scores)
+        .map((address, i) => [address, String(updated[i * 2][1])])
+    )
+  )
+}
+
+async function handlePaidScheduledRewards (req, res, redis, signerAddresses) {
+  const body = JSON.parse(await getRawBody(req, { limit: '1mb' }))
+
+  httpAssert(
+    typeof body === 'object' && body !== null,
+    400,
+    'Request body should be an object'
+  )
+  httpAssert(
+    typeof body.rewards === 'object' && body.rewards !== null,
+    400,
+    '.rewards should be an object'
+  )
+  httpAssert(
+    Object.keys(body.rewards).every(ethers.isAddress),
+    400,
+    'All .rewards keys should be 0x addresses'
+  )
+  httpAssert(
+    Object.values(body.rewards).every(score => {
+      try {
+        return BigInt(score) > 0n
+      } catch {
+        return false
+      }
+    }),
+    400,
+    'All .rewards values should be positive numbers encoded as string'
+  )
+
+  validateSignature(body.signature, body.rewards, signerAddresses)
+
+  const timestamp = new Date()
+  const tx = redis.multi()
+  for (const [address, amount] of Object.entries(body.rewards)) {
+    tx.hincrby('rewards', address, BigInt(amount) * -1n)
+    tx.rpush(
+      'log',
+      JSON.stringify({
+        timestamp,
+        address,
+        scheduledRewards: String(BigInt(amount) * -1n)
```

Comment:
Same comment as above: When I read the docs, I assumed `scheduledRewards` contained the new amount after the round rewards were applied. It took me a while to notice that the log tells us only the delta. Let's find a better name for this field, please.

Reply:
See above